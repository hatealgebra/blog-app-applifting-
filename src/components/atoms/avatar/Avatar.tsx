import React from 'react';

import StyledAvatar from './avatar.styled';
import avatarGeneric from '../../../images/avatar-fallback.png';
import { AvatarProps } from './avatar.typed';

const Avatar = ({ size = 'lg', src, alt }: AvatarProps) => {
  return (
    <StyledAvatar
      className="avatar"
      size={size}
      src={src || avatarGeneric}
      alt={alt || 'fallback user avatar'}
    />
  );
};

export default Avatar;
