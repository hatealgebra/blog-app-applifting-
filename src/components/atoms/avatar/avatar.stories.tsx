import React from 'react';
import Avatar from './Avatar';

import userAvatarPhoto from '../../../images/avatar-mock.jpg';
import { ComponentMeta } from '@storybook/react';

export const DifferentImages = () => {
  return (
    <>
      <Avatar src={userAvatarPhoto} alt="Generic user avatar" />
      <Avatar />
    </>
  );
};

export const AllSizes = () => {
  return (
    <>
      <Avatar size="lg" src={userAvatarPhoto} alt="Generic user avatar" />
      <Avatar size="md" src={userAvatarPhoto} alt="Generic user avatar" />
      <Avatar size="sm" src={userAvatarPhoto} alt="Generic user avatar" />
    </>
  );
};

export default {
  title: 'Atoms/Avatar',
  component: Avatar,
  decorators: [
    (Story) =>
      (
        <div style={{ display: 'flex', gap: '5px', alignItems: 'flex-end' }}>
          <Story />
        </div>
      ) as ComponentMeta<typeof Avatar>,
  ],
};
