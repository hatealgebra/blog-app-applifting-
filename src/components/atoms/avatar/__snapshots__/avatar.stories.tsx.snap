// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/Avatar AllSizes smoke-test 1`] = `
<div style="display: flex; gap: 5px; align-items: flex-end;">
  <img class="sc-kDnyCx dyuMnk avatar"
       src="static/media/src/images/avatar-mock.jpg"
       alt="Generic user avatar"
  >
  <img class="sc-kDnyCx enRCkM avatar"
       src="static/media/src/images/avatar-mock.jpg"
       alt="Generic user avatar"
  >
  <img class="sc-kDnyCx dmEavk avatar"
       src="static/media/src/images/avatar-mock.jpg"
       alt="Generic user avatar"
  >
</div>
`;

exports[`Atoms/Avatar DifferentImages smoke-test 1`] = `
<div style="display: flex; gap: 5px; align-items: flex-end;">
  <img class="sc-kDnyCx dyuMnk avatar"
       src="static/media/src/images/avatar-mock.jpg"
       alt="Generic user avatar"
  >
  <img class="sc-kDnyCx dyuMnk avatar"
       src="static/media/src/images/avatar-fallback.png"
       alt="fallback user avatar"
  >
</div>
`;
